platform/fabricx/core/finality/nlm_test.go

```diff
@@ -56,26 +56,65 @@ func (m *mockListener) getStatus() (string, int) {
 	return m.txID, m.status
 }
 
-func setupTest(tb testing.TB) (*notificationListenerManager, *mock.FakeNotifier_OpenNotificationStreamClient, context.Context) {
+func setupTest(tb testing.TB) (*notificationListenerManager, *mock.FakeNotifier_OpenNotificationStreamClient, context.CancelFunc) {
 	tb.Helper()
 
-	ctx := tb.Context()
+	// Create the context that will govern the manager's lifetime
+	managerCtx, cancelFunc := context.WithCancel(context.Background())
 
 	fakeStream := &mock.FakeNotifier_OpenNotificationStreamClient{}
-	fakeStream.ContextReturns(ctx)
+	// The stream's context is what the errgroup will derive its context from
+	fakeStream.ContextReturns(managerCtx)
 
 	nlm := &notificationListenerManager{
 		notifyStream:  fakeStream,
 		requestQueue:  make(chan *protonotify.NotificationRequest, 1),
 		responseQueue: make(chan *protonotify.NotificationResponse, 1),
 		handlers:      make(map[driver.TxID][]fabric.FinalityListener),
+		cancelFunc:    cancelFunc, // Store the function to stop the manager
 	}
 
-	return nlm, fakeStream, ctx
+	return nlm, fakeStream, cancelFunc
+}
+
+// runManager starts the manager asynchronously and ensures Stop() is called on test cleanup.
+func runManager(t *testing.T, nlm *notificationListenerManager, managerCancel context.CancelFunc) {
+	t.Helper()
+
+	// Start the manager asynchronously
+	errs := make(chan error, 1)
+	go func() {
+		errs <- nlm.Start()
+	}()
+
```

Comment:
that is an interesting test. I think it is not guaranteed that `nlm.Start()` is actually called. The go rountine might not start in time, so `t.Cleanup()` gets called before I believe!?!?